api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -903,6 +903,10 @@ type AzureNodePoolPlatform struct {
 	// +kubebuilder:default:=default
 	// +kubebuilder:validation:Required
 	SubnetName string `json:"subnetName"`
+	// Diagnostics specifies the diagnostics settings for a virtual machine.
```

Comment:
>/ If not specified then Boot diagnostics (Managed) will be enabled.

Should we make that a explicit default for ux and to avoid potential transparent breaking behaviour?